lib/lumberjack/beats/server.rb

```diff
@@ -139,13 +130,80 @@ def close
       @close.make_true
       @server.close unless @server.closed?
     end
-  end # class Server
+
+    private
+    def retrieve_current_port
+      # Query the port in case the port number is '0'
+      # TCPServer#addr == [ address_family, port, address, address ]
+      @server.addr[1]
+    end
+
+    def certificate_authorities
+      Array(@options[:ssl_certificate_authorities])
+    end
+
+    def server_private_key
+      OpenSSL::PKey::RSA.new(File.read(@options[:ssl_key]),
```

Comment:
So this method will return nil if `ssl_key` is not provided? Should it be an exception if `server_private_key` is invoked but `:ssl_key` is not provided?